src/decoration.coffee

```diff
@@ -46,6 +47,20 @@ class Decoration
     else
       type is decorationProperties.type
 
+  # Private: Check if the `decorationProperties.trueType` matches `type`
+  #
+  # * `decorationProperties` {Object} eg. `{type: 'line-number', class: 'my-new-class'}`
+  # * `type` {String} type like `'gutter'`, `'line'`, etc. `type` can also
+  #   be an {Array} of {String}s, where it will return true if the decoration's
+  #   type matches any in the array.
+  #
+  # Returns {Boolean}
+  @isTrueType: (decorationProperties, type) ->
```

Comment:
Why not bake the special case for `line-number` into the existing `isType` method? You could just always say that `line-number` decorations matched the type of `gutter`.